Tests/KituraTests/TestCodableRouter.swift

```diff
@@ -726,6 +735,19 @@ class TestCodableRouter: KituraTest {
             .hasStatus(.badRequest)
             .hasNoData()
 
+            .request("get", path: "/optionalquery\(queryStr)")
+            .hasStatus(.OK)
+            .hasContentType(withPrefix: "application/json")
+            .hasData(expectedQuery)
+
+            .request("get", path: "/optionalquery")
+            .hasStatus(.OK)
+            .hasNoData()
+
+            .request("get", path: "/optionalquery?param=badRequest")
+            .hasStatus(.badRequest)
+            .hasNoData()
```

Comment:
I feel there are too many things being tested in one test function.  I appreciate there's precedent, but I'd prefer a separate test such as `testCodableGetSingleOptionalQueryParameters()`.

Because these tests make their assertions within the KituraTestBuilder functions, when one fails you only get the failing line within the helper (eg. `hasStatus`) function, and not the line of the `testXyz()` function that called it.  If each test only has one instance of `hasStatus(.OK)` then it's easy to tell what failed, but in this instance we are making the same assertion in 3 places.